tests/unit/s2n_random_test.c

```diff
@@ -142,6 +146,110 @@ static S2N_RESULT s2n_basic_pattern_tests(S2N_RESULT (*s2n_get_random_data_cb)(s
     return S2N_RESULT_OK;
 }
 
+static void swap(size_t i, size_t j,
+    uint64_t range_results[NUMBER_OF_RANGE_FUNCTION_CALLS]) {
+    uint64_t temp = range_results[j];
+    range_results[j] = range_results[i];
+    range_results[i] = temp;
+}
+
+static void sort_array(uint64_t range_results[NUMBER_OF_RANGE_FUNCTION_CALLS]) {
```

Comment:
Implementing a sort method (and a swap method) just for a test seems unnecessary, and potentially dangerous if your test relies on it actually always sorting correctly.

Can you just use qsort?